src/radical/pilot/utils/prof_utils.py

```diff
@@ -199,6 +199,114 @@
+        'u_agent_cu_pre_execute'       : [ {ru.EVENT: 'cu_pre_start'    , ru.STATE: None                            },
+                                           {ru.EVENT: 'cu_pre_stop'     , ru.STATE: None                            }  ],
+        'u_agent_cu_execute_start'     : [ {ru.EVENT: 'cu_pre_stop'     , ru.STATE: None                            },
+                                           {ru.EVENT: 'cu_exec_start'   , ru.STATE: None                            }  ],
+        'u_agent_cu_execute'           : [ {ru.EVENT: 'cu_exec_start'   , ru.STATE: None                            },
+                                           {ru.EVENT: 'cu_exec_stop'    , ru.STATE: None                            }  ],
+        'u_agent_cu_stop'              : [ {ru.EVENT: 'cu_exec_stop'    , ru.STATE: None                            },
+                                           {ru.EVENT: 'cu_stop'         , ru.STATE: None                            }  ],
+        'u_agent_cu_unschedule_start'  : [ {ru.EVENT: 'exec_stop'       , ru.STATE: None                            },
+                                           {ru.EVENT: 'unschedule_start', ru.STATE: None                            }  ], # PROBLEM: discontinuity
+        'u_agent_cu_stage_out_start'   : [ {ru.EVENT: 'unschedule_start', ru.STATE: None                            },
+                                           {ru.EVENT: 'state'           , ru.STATE: rps.AGENT_STAGING_OUTPUT_PENDING}  ],
+        'u_agent_cu_stage_out_queue'   : [ {ru.EVENT: 'state'           , ru.STATE: rps.AGENT_STAGING_OUTPUT_PENDING},
+                                           {ru.EVENT: 'state'           , ru.STATE: rps.AGENT_STAGING_OUTPUT        }  ],
+        'u_agent_cu_stage_out'         : [ {ru.EVENT: 'state'           , ru.STATE: rps.AGENT_STAGING_OUTPUT        },
+                                           {ru.EVENT: 'state'           , ru.STATE: rps.UMGR_STAGING_OUTPUT_PENDING }  ],
+        'u_agent_cu_unschedule_stop'   : [ {ru.EVENT: 'state'           , ru.STATE: rps.UMGR_STAGING_OUTPUT_PENDING },
+                                           {ru.EVENT: 'unschedule_stop' , ru.STATE: None                            }  ],
+        'u_agent_cu_push_to_umngr'     : [ {ru.EVENT: 'unschedule_stop' , ru.STATE: None                            },
+                                           {ru.EVENT: 'state'           , ru.STATE: rps.UMGR_STAGING_OUTPUT         }  ], # push/pull mongodb
+        'u_umngr_cu_destroy'           : [ {ru.EVENT: 'state'           , ru.STATE: rps.UMGR_STAGING_OUTPUT         },
+                                          [{ru.EVENT: 'state'           , ru.STATE: rps.DONE                        },
+                                           {ru.EVENT: 'state'           , ru.STATE: rps.CANCELED                    },
+                                           {ru.EVENT: 'state'           , ru.STATE: rps.FAILED                      } ]]
+    }
+} 
+
+
```

Comment:
I think I disagree with the unschedule parts.  My suggestion would be to create a continuous line without the `unschedule_*` events, and add `u_agent_unschedule` *additionally*, since the unschedule activity happens really concurent to the task's state progression (and the normal event timeline).  This also ensures consistentcy: the above can lead to negative durations on a fast unschedule for example.

Reply:
Done.